Tools/scripts/pathfix.py

```diff
@@ -164,12 +181,47 @@ def fix(filename):
     # Return success
     return 0
 
+
+def parse_shebang(shebangline):
+    """takes shebangline and returns tuple with string of shebang's flags
+    and string of shebang's args
+    parse_shebang(b'#!/usr/bin/python -f sfj') --> (b'f',b'sfj')
+    """
+    end = len(shebangline)
+    start = shebangline.find(b' -', 0, end) + 2  # .find() returns index at space
+    if chr(shebangline[start]).isalpha():
+        flags = shebangline[start:end].split()
+        args = b''
+        if len(flags) > 1:
+            args = b' '.join(flags[1:])
+        flags = flags[0]
+        return flags, args
+    return b'', b''
+
+
 def fixline(line):
     if not line.startswith(b'#!'):
         return line
+
     if b"python" not in line:
         return line
-    return b'#! ' + new_interpreter + b'\n'
+
+    fixedline = b'#! ' + new_interpreter
+    if add_flag is None:
+        return fixedline + b'\n'
+
+    flags, args = parse_shebang(line)
+    if args:
+        args = b' ' + args
+
+    if flags == b'' and add_flag == b'':
+        return fixedline + b'\n'
+
+    if add_flag in flags:
+        return fixedline + b' -' + flags + args + b'\n'
+
+    return fixedline + b' -' + add_flag + flags + args + b'\n'
```

Comment:
I would expect a line like: `b'#! ' + new_interpreter + flags + b'\n'`. Then arrange your code to build flags as you want, flags can an empty string.

I also suggest to create a sub-function fix_shebang() to clarify the purpose of the function.